docs/concepts/what-is-a-run-id.md

```diff
@@ -16,3 +16,11 @@ For example, a Temporal Cron Job is a chain of Workflow Executions that all have
 Each Workflow Execution within the chain is considered a _Run_.
 
 A Run Id uniquely identifies a Workflow Execution even if it shares a Workflow Id with other Workflow Executions.
+
+:::note
+
+The current Run Id is mutable and can change during a Workflow Retry. Because of that, you should not rely on storing the current Run Id, or using it for any logical choices, as a Workflow Retry changes the Run Id and can lead to non-determinism issues,
```

Comment:
```suggestion
Don't rely on storing the current Run Id or using it for any logical choices.
A Workflow Retry changes the Run Id.
Because the current Run Id, is mutable, relying on it might produce non-determinism issues,
```